Snowflake.Data/Core/SFResultSet.cs

```diff
@@ -99,23 +154,64 @@ internal override bool Next()
                 throw new SnowflakeDbException(SFError.DATA_READER_ALREADY_CLOSED);
             }
 
+            if (0 == fetchCount)
+            {
+                Logger.Info("Starting. Fetching first row.");
+                stopwatch.Restart();
+            }
+            else if (0 == (fetchCount % 100000))
+            {
+                LogAverageConversionTimesForLast100000rows();
+            }
+
+            fetchCount++;
             _currentChunkRowIdx++;
             if (_currentChunkRowIdx < _currentChunkRowCount)
             {
+                if (0 == (fetchCount % 100000))
+                {
+                    Logger.Info($"Fetched {fetchCount} rows");
+                }
                 return true;
             }
 
             if (_chunkDownloader != null)
             {
                 Logger.Info("Get next chunk from chunk downloader");
-                IResultChunk nextChunk = Task.Run(async() => await _chunkDownloader.GetNextChunkAsync()).Result;
+                IResultChunk nextChunk = Measure(
+                        @"nextChunk",
+                        out double nextChunkTime,
+                        () =>
+                        {
+                            return Task.Run(async () => await _chunkDownloader.GetNextChunkAsync().ConfigureAwait(false)).Result;
+                        });
+
                 if (nextChunk != null)
                 {
                     resetChunkInfo(nextChunk);
+                    if (0 == (fetchCount % 100000))
+                    {
+                        Logger.Info($"Fetched {fetchCount} rows");
+                    }
                     return true;
                 }
             }
-           return false;
+
+            stopwatch.Stop();
+            Logger.Debug("All data retrieved. Last row returned.");
+            Logger.Debug($"Fetch process took {stopwatch.Elapsed.TotalMilliseconds} ms");
+            if (Logger.IsDebugEnabled())
+            {
+                if (timings.TryGetValue("nextChunk", out double nextChunkTotalTime))
+                {
+                    Logger.Debug("---- Time spent getting next chunk ----");
+                    Logger.Debug($"Total time : {nextChunkTotalTime} ms");
+                    Logger.Debug($"Average time : {nextChunkTotalTime / (float)_totalChunkCount } ms\n");

```

Comment:
Can you change those logging to debug level? This way potentially we can get those data points only if we enabled the info level logging, which potentially have a much less log volume.

Reply:
The logging is already in Debug level here. The ones at info level are not printing much, only a line at the beginning of the read, another one every 100000 rows and a last one when retrieval is done.

Reply:
The logging is already in Debug level here. The ones at info level are not printing much, only a line at the beginning of the read, another one every 100000 rows and a last one when retrieval is done.

The timing processing is done if log level is Debug only.

This is a sample for select * from SNOWFLAKE_SAMPLE_DATA.TPCH_SF100.LINEITEM limit 350000:
- ERROR - No log : Fetch time = 9157
- INFO - Will log start/end fetch, every 100000 rows and total fetch time : Fetch time = 9797
- DEBUG - Same as INFO plus timing for the methods GetValue, ConvertToCSharpVal, getObjectInternal and GetNextChunkAsync : Fetch time = 15407